File: app/Http/Requests/Api/V1/UpdateStoreRequest.php
<?php

namespace App\Http\Requests\Api\V1;

use Illuminate\Foundation\Http\FormRequest;

/**
 * Request validation for updating store information.
 *
 * Used by store owners to update their store's basic details.
 */
class UpdateStoreRequest extends FormRequest
{
    /**
     * Determine if the user is authorized to make this request.
     *
     * Authorization is handled in the controller using isOwnerOf() check.
     */
    public function authorize(): bool
    {
        return true;
    }

    /**
     * Get the validation rules that apply to the request.
     *
     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
     */
    public function rules(): array
    {
        return [
            'name' => ['sometimes', 'required', 'string', 'max:255'],
            'description' => ['sometimes', 'nullable', 'string', 'max:2000'],
            'city' => ['sometimes', 'nullable', 'string', 'max:100'],
        ];
    }

    /**
     * Get custom error messages for validation rules.
     *
     * @return array<string, string>
     */
    public function messages(): array
    {
        return [
            'name.required' => 'The store name is required.',
            'name.max' => 'The store name cannot exceed 255 characters.',
            'description.max' => 'The description cannot exceed 2000 characters.',
            'city.max' => 'The city name cannot exceed 100 characters.',
        ];
    }
}

File: ios/Preuvely/Preuvely/Models/Requests/UpdateStoreRequest.swift
import Foundation

// MARK: - Update Store Request

/// Request model for updating store information
struct UpdateStoreRequest: Encodable {
    let name: String?
    let description: String?
    let city: String?

    /// Creates an update store request with optional fields
    /// - Parameters:
    ///   - name: The new store name (optional)
    ///   - description: The new store description (optional)
    ///   - city: The new store city (optional)
    init(name: String? = nil, description: String? = nil, city: String? = nil) {
        self.name = name
        self.description = description
        self.city = city
    }
}
